### PR TITLE
Catch hover on as const

### LINKED ISSUE #43515: Hovering `as const` returns `any`
TS stable and nightly
[playground](https://www.typescriptlang.org/play?#code/ETAEEMGdQYwewHaQC5A)
```ts
"" as const
```
chrome_2021-04-03_13-27-01

### PR DESCRIPTION
**Pull request by kunaljaydesai**
Fixes #43515 

First PR here, would appreciate some help!

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -15636,6 +15636,11 @@ namespace ts {
                 case SyntaxKind.Identifier as TypeNodeSyntaxKind:
                 case SyntaxKind.QualifiedName as TypeNodeSyntaxKind:
                 case SyntaxKind.PropertyAccessExpression as TypeNodeSyntaxKind:
+                    if (node.kind === SyntaxKind.Identifier as TypeNodeSyntaxKind &&
+                        node.parent.kind === SyntaxKind.TypeReference &&
+                        (<Identifier>(node as Node)).originalKeywordKind === SyntaxKind.ConstKeyword) {
+                        return errorType;
```
Comment (kunaljaydesai) [inline src/compiler/checker.ts:15642]: This isn't the right type to return, what would be?

**Comment by Kingwl** [comment]
I think this is not an issue of type checker.

May consider about `src\services\services.ts@getNodeForQuickInfo`. (Add some special case here).

**Comment by kunaljaydesai** [comment]
thanks for the pointers @Kingwl! I've updated the PR to reflect that and added a test

another option to consider is maybe `isConstTypeReference(node.parent)` in `shouldGetType`:
Screen Shot 2021-04-07 at 11 38 41 PM

let me know which is preferred!

**Comment by Kingwl** [comment]
I felt ok with both of them options.

You could modify the `getNodeForQuickInfo` If you wanna provide some quick info (eg: `node as type` shows node if you hover to type).

But If you want to keep it silent, `shouldGetType` might be a better choice.

**Comment by kunaljaydesai** [comment]
I think I'll keep it silent for now so opted for the second approach! Let me know if there's anything else needed to get the PR merged

File: tests/cases/fourslash/quickInfoForConstTypeReference.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/// <reference path="fourslash.ts" />
+
+////"" as /**/const;
+
+verify.quickInfoAt("", "");
```
Comment (Kingwl) [inline tests/cases/fourslash/quickInfoForConstTypeReference.ts:5]: Use `verify.not.quickInfoExists()` instead.

**Comment by Kingwl** [comment]
BTW: It's no need to force push. We usually merge the PR with squash.

File: tests/cases/fourslash/quickInfoForConstTypeReference.ts (hunk 2 above)
Comment (kunaljaydesai) [reply tests/cases/fourslash/quickInfoForConstTypeReference.ts]: updated!

**Review by Kingwl — APPROVED** [review]
Looks good! Thanks for your contribution.

**Review by orta — APPROVED** [review]
Agree, thanks!

**Pull request merged**